scripts/NameGenerator/NameGenerator.gml

```diff
@@ -0,0 +1,6 @@
+// Script assets have changed for v2.3.0 see
+// https://help.yoyogames.com/hc/en-us/articles/360005277377 for more information
+function NameGenerator() constructor
+{
+	// TODO, thinking how to make name generation bit more generic
```

Comment:
95% of the code is same in StarNameGenerator and SystemNameGenerator, need to think a bit how to make it more generic

Reply:
```suggestion
    static get_random_name = function() {
        return array_pop(names)
    }
    names = array_shuffle(names_array)
    original_names = []
    array_copy(original_names, 0, names, 0)
```
this uses the idea of passing in the loaded JSON rather than a filepath, otherwise the loaded json could just be loaded into `original_names` as a cache and then names could be set to `array_shuffle(original_names)`

Reply:
found a way, basically I split name loading from jsons and name composition into different functions.